frontend/src/app/application-forms/tree-application-form/tree-application-form.component.ts

```diff
@@ -104,10 +97,23 @@ export class TreeApplicationFormComponent implements OnInit {
     }
   }
 
+  rePopulateForm() {
+    this.applicationForm.get('firstName').setValue(this.permit.firstName);
+    this.applicationForm.get('lastName').setValue(this.permit.lastName);
+    this.applicationForm.get('emailAddress').setValue(this.permit.emailAddress);
+    this.applicationForm.get('quantity').setValue(this.permit.quantity);
+    this.quantityChange(this.permit.quantity);
+  }
   createApplication() {
```

Comment:
We cleaned this up so that the method to remove the old permit is moved out of the complete function so we were able to remove the conditional. This is now more readable.